operators/pkg/controller/elasticsearch/driver/downscale_test.go

```diff
@@ -679,13 +680,151 @@ func Test_doDownscale_zen2VotingConfigExclusions(t *testing.T) {
 			// check call to zen2 is the expected one
 			require.Equal(t, tt.wantZen2Called, esClient.AddVotingConfigExclusionsCalled)
 			require.Equal(t, tt.wantZen2CalledWith, esClient.AddVotingConfigExclusionsCalledWith)
+			// check zen1 was not called
+			require.False(t, esClient.SetMinimumMasterNodesCalled)
 		})
 	}
 }
 
-func Test_doDownscale_callsZen1ForMasterNodes(t *testing.T) {
-	// TODO: implement with https://github.com/elastic/cloud-on-k8s/issues/1281
-	//  to handle the 2->1 masters case
+func Test_doDownscale_zen1MinimumMasterNodes(t *testing.T) {
+	es := v1alpha1.Elasticsearch{ObjectMeta: metav1.ObjectMeta{Namespace: sset3Replicas.Namespace, Name: "es"}}
+	ssetMasters := nodespec.CreateTestSset("masters", "6.8.0", 3, true, false)
+	masterPods := []corev1.Pod{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: sset3Replicas.Namespace,
+				Name:      sset3Replicas.Name + "-0",
+				Labels: map[string]string{
+					label.StatefulSetNameLabelName:         sset3Replicas.Name,
+					label.ClusterNameLabelName:             es.Name,
+					string(label.NodeTypesMasterLabelName): "true",
+				},
+			},
+			Status: corev1.PodStatus{
+				Conditions: []corev1.PodCondition{
+					{
+						Status: corev1.ConditionTrue,
+						Type:   corev1.ContainersReady,
+					},
+					{
+						Status: corev1.ConditionTrue,
+						Type:   corev1.PodReady,
+					},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: sset3Replicas.Namespace,
+				Name:      sset3Replicas.Name + "-1",
+				Labels: map[string]string{
+					label.StatefulSetNameLabelName:         sset3Replicas.Name,
+					label.ClusterNameLabelName:             es.Name,
+					string(label.NodeTypesMasterLabelName): "true",
+				},
+			},
+			Status: corev1.PodStatus{
+				Conditions: []corev1.PodCondition{
+					{
+						Status: corev1.ConditionTrue,
+						Type:   corev1.ContainersReady,
+					},
+					{
+						Status: corev1.ConditionTrue,
+						Type:   corev1.PodReady,
+					},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: sset3Replicas.Namespace,
+				Name:      sset3Replicas.Name + "-2",
+				Labels: map[string]string{
+					label.StatefulSetNameLabelName:         sset3Replicas.Name,
+					label.ClusterNameLabelName:             es.Name,
+					string(label.NodeTypesMasterLabelName): "true",
+				},
+			},
+			Status: corev1.PodStatus{
+				Conditions: []corev1.PodCondition{
+					{
+						Status: corev1.ConditionTrue,
+						Type:   corev1.ContainersReady,
+					},
+					{
+						Status: corev1.ConditionTrue,
+						Type:   corev1.PodReady,
+					},
+				},
+			},
+		},
+	}
+	ssetData := nodespec.CreateTestSset("datas", "6.8.0", 3, false, true)
```

Comment:
If only we could name function fields 😄 
Instead of creating  4 additional functions here, maybe we could just provide a struct?

```
nodespec.CreateTestSset(nodespec.CreateTestSsetOptions{Name: "data", Version: "6.8.0", Replicas: 3,  Master: true})
```

Your version looks easier to read but would probably look more like this when used from another pkg:
```
nodespec.CreateTestSset("datas", nodespec.Version("6.8.0"), nodespec.Replicas(3), nodespec.Master(true), nodespec.Data(true))
```

Reply:
I'll go with the following approach:
```go
nodespec.TestSset{Name: "master", Version: "7.1.0", Replicas: 3, Master: true, Data: false}.Build()
```